roles/ldap/tasks/main.yml

```diff
@@ -8,6 +8,11 @@
       - openldap-clients
       - nss-pam-ldapd
       - openssh-ldap
+      - python36
```

Comment:
Do we really need python 3.6 for such a small script? If something now goes wrong with deploying/updating that extra python36, you will lock yourself out. Would be better if this does not use any special stuff and works with both the default python2.x and 3.x....

Reply:
Python2 will retire in 3 months. (see https://pythonclock.org/)
Furthermore python3.6 is in the standard repositories so i see no reason not to rely on it.

Reply:
I know and I'd rather see Python 2 removed all together, but the default Python on CentOS 7.x is still 2.7. Python 3.6 may be available from default repos, but is not by default installed on the CentOS 7.x cloud image and nor does it use the default interpreter name for python: ```/usr/bin/env python``` will fail :(
For something this crucial I'd rather see no unneccessary dependencies... It's great the script is already compatible with Python 3, but can't we simply make sure this runs on the default python too?

Reply:
With 2 minor changes the python script can be made compatible with python 2.x except for the missing ```sshpubkeys``` package. This:
```
- name: Install dependencies of ssh wrapper.
  pip:
    name: sshpubkeys
    executable: pip3
```
is something I'd consider very dangerous as we now have a external dependency on pypi.org: this code is not fetched from the spacewalk server and is not under version control. Deploying this a day later may result in a different and potentially broken version.

Reply:
> I know and I'd rather see Python 2 removed all together, but the default Python on CentOS 7.x is still 2.7. Python 3.6 may be available from default repos, but is not by default installed on the CentOS 7.x cloud image and nor does it use the default interpreter name for python: `/usr/bin/env python` will fail :(
> For something this crucial I'd rather see no unneccessary dependencies... It's great the script is already compatible with Python 3, but can't we simply make sure this runs on the default python too?
Of course we can, it would probably work out of the box though. I just don't follow your reasoning. And think it would be a bad idea to build something on python2 in 2019.

Python 3 is installed before the script from the repo's . It won't magically disappear from a system.

Reply:
> - name: Install dependencies of ssh wrapper.
>   pip:
>     name: sshpubkeys
>     executable: pip3
> ```
> 
> is something I'd consider very dangerous as we now have a external dependency on pypi.org: this code is not fetched from the spacewalk server and is not under version control. Deploying this a day later may result in a different and potentially broken version.

You have a point here. I could try and convert the pypi package to a RPM package and host that on our spacewalk. I prefer to keep using a tried, tested and maintained ssh key parser over something bodged together ourselves.